packages/flame/test/components/priority_test.dart

```diff
@@ -163,10 +164,13 @@ void main() {
         c1.priority = 10;
         a2.priority = 0;
         game.update(0);
+        await game.ready();
 
         a.assertCalled(1);
         b.assertCalled(0);
-        c.assertCalled(1);
+        // TODO(luan): consider optimizing so that multiple priority changes
+        // on the same parent on the same tick only trigger one reorder call
```

Comment:
Didn't we use to have this? 🤔

Reply:
yes the previous implementation did this naturally due to how it was setup

Reply:
It will be quite expensive to not have that if there are a lot of priority changes, any ideas of how this could be implemented with the new way?

Reply:
that is only if you have multiple priority changes on the same parent on the same tick, so I don't think it is super relevant. we can eventually revamp the whole lifecycle queue to re-group and optimize events in general between the 4 types

Reply:
I remember that there were complaints about it before we implemented it. Since we don't have any other way to do z-index, priority is heavily used like that in for example isometric games, it would go from O(n) to O(n²).

Reply:
That would be `O(n^2)` only if you are changing the order of every single component within a parent on every single tick, which I think is unrealistic (and already would not perform well regardless). But I changed the implementation to batch reorders, to the cost of some minor overhead for the common case.

Reply:
When you measure big O you do it for the asymptotic upper bound, hence `O(n²)`.
The point is, just reordering many (N) per tick would be N times as expensive, which was causing the problem I mentioned previously.
The updated solution looks good!